src/util/misc/matrix.ts

```diff
@@ -0,0 +1,209 @@
+import { cos } from "./cos";
+import { sin } from "./sin";
+import { degreesToRadians } from "./radiansDegreesConversion";
+import { iMatrix, PiBy180 } from "../../constants";
+import { IPoint, Point } from "../../point.class";
+import { TDegree, TMat2D } from "../../typedefs";
+
+type TRotateMatrixArgs = {
+  angle?: TDegree;
+}
+
+type TTranslateMatrixArgs = {
+  translateX?: number;
+  translateY?: number;
+}
+
+type TScaleMatrixArgs = {
```

Comment:
type TScaleMatrixArgs = Patial<{ ... }>

https://www.typescriptlang.org/docs/handbook/utility-types.html

Reply:
Why you prefer Partial here?